// yuuta-functions/functions.js
const api = require('./api.json')
const db = require('quick.db')

function abreviar(number=1, precision=2) {
    const suffsFromZeros = { 0:'', 3:'k', 6:'M', 9:'B', 12:'T', 15:'Qa', 18:'Qi', 21:'Sx' }
    const { length } = number.toString()
    const lengthThird = length%3
    const divDigits = length-(lengthThird || lengthThird+3)
    const calc = ''+(number/(10**divDigits)).toFixed(precision)
  
    return number < 1000 ? ''+number : (calc.indexOf('.') === calc.length-3 ? calc.replace(/\.00/, '') : calc)+suffsFromZeros[divDigits]
  }

function delay(time=1, f) {
    return setTimeout(() => {}, time, f)
}

function sort(o) {
    return o[Math.floor(Math.random() * o.length)]
}

function cry() {
    return sort(api.cry)
}
function kiss() {
    return sort(api.kiss)
}
function slap() {
    return sort(api.slap)
}
function hug() {
    return sort(api.hug)
}
function pat() {
    return sort(api.pat)
}
function poke() {
    return sort(api.poke)
}
function tickle() {
    return sort(api.tickle)
}
function feed() {
    return sort(api.feed)
}
function cuddle() {
    return sort(api.cuddle)
}

function set(key, value) {
    db.set(key, value)
}

function get(key) {
    return db.get(key)
}

function pull(key, value, once = false) {
    db.pull(key, value, once)
}

function add(key, value) {
    db.add(key, value)
}

function remove(key, value) {
    db.delete(key, value)   
}
function removeAll() {
    db.deleteAll()
}
function push(key, value) {
    db.push(key, ...values)
}
function has(key) {
    db.has(key)
}
function startsWith(query, key = "") {
    return db.startsWith(query, key = "")
}

module.exports = {
    srt: sort,
    chorar: cry,
    beijar: kiss,
    tapa: slap,
    abraço: hug,
    cafune: pat,
    cutucar: poke,
    cocegas: tickle,
    comida: feed,
    mimar: cuddle,
    ab: abreviar,
    delay,
    set,
    get,
    push,
    pull,
    add,
    remove,
    removeAll,
    has,
    startsWith
}
